app/src/main/java/com/android/voyageur/App.kt

```diff
@@ -34,7 +35,12 @@ fun VoyageurApp(placesClient: PlacesClient) {
   val navController = rememberNavController()
   val navigationActions = NavigationActions(navController)
   val tripsViewModel: TripsViewModel = viewModel(factory = TripsViewModel.Factory)
-  val userViewModel: UserViewModel = viewModel(factory = UserViewModel.Factory)
+  // Get the current context from the Compose hierarchy
+  val context = LocalContext.current
+
+  // Use UserViewModel.provideFactory(context) instead of UserViewModel.Factory
+  val userViewModel: UserViewModel = viewModel(factory = UserViewModel.provideFactory(context))
```

Comment:
why do we need the context in the userviewmodel? The view model should be agnostic of context which is related to the UI / composables.

Reply:
You are completely right! I updated the code to include 2 compostables that can be called in order for the use model to be agnostic to the context. Thanks for catching that!